src/plugins/inspector/public/views/requests/components/requests_view.tsx

```diff
@@ -19,47 +19,53 @@
 
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {
-  EuiEmptyPrompt,
-  EuiSpacer,
-  EuiText,
-  EuiTextColor,
-} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiEmptyPrompt, EuiSpacer, EuiText, EuiTextColor } from '@elastic/eui';
 
-import { RequestStatus } from 'ui/inspector/adapters';
+import { RequestStatus } from '../../../adapters';
+import { Request } from '../../../adapters/request/types';
+import { InspectorViewProps } from '../../../types';
 
 import { RequestSelector } from './request_selector';
 import { RequestDetails } from './request_details';
-import { FormattedMessage } from '@kbn/i18n/react';
-import { i18n } from '@kbn/i18n';
 
-class RequestsViewComponent extends Component {
+interface RequestSelectorState {
+  requests: Request[];
+  request: Request;
+}
+
+export class RequestsViewComponent extends Component<InspectorViewProps, RequestSelectorState> {
+  static propTypes = {
+    adapters: PropTypes.object.isRequired,
+  };
 
-  constructor(props) {
+  constructor(props: InspectorViewProps) {
```

Comment:
Question: is `InspectorViewProps` still needed here? I'd assume generic above (`Component<InspectorViewProps, RequestSelectorState>`) would already populate this for you, but that might be wrong

Reply:
![image](https://user-images.githubusercontent.com/20072247/63159201-97856180-c023-11e9-8ec3-9df7c176d90a.png)